src/iosrv/srv.c

```diff
@@ -266,6 +354,7 @@ dss_sched_run(ABT_sched sched)
 			}
 			work_count = 0;
 			ABT_xstream_check_events(sched);
+			ds_check_expire_list(false);
```

Comment:
Shouldn't we do this check every time before calling dss_sched_unit_pop()? And I don't quite see how dss_sched_unit_pop() will deal with these sleeping ULTs: dss_sched_unit_pop() choose a pool based on the ABT_pool_get_total_size(), I think this total size includes the sleeping ULTs, right? Then will unit_pop() pop out a sleeping ULT?

Reply:
oh, the pool is already suspended, so it will not be added to the pool. so pop will not popup "sleeping" ULT.

Reply:
right, I meant we should use ABT_pool_get_size() but not ABT_pool_get_total_size(), the latter one returned the number including blocked ULTs.